fwoa-tools/src/migrationUtils.test.ts

```diff
@@ -0,0 +1,156 @@
+/*
+ *  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *  SPDX-License-Identifier: Apache-2.0
+ */
+import { createWriteStream, WriteStream } from 'fs';
+import AWS from 'aws-sdk';
+import { InitiateAuthRequest } from 'aws-sdk/clients/cognitoidentityserviceprovider';
+import * as AWSMock from 'aws-sdk-mock';
+import { checkConfiguration, getFhirClient, getFhirClientSMART } from './migrationUtils';
+
+const env = process.env;
+AWSMock.setSDKInstance(AWS);
+
+describe('MigrationUtils', () => {
+  beforeEach(() => {
+    expect.hasAssertions();
+    AWSMock.restore();
+  });
+
+  afterEach(() => {
+    AWSMock.restore();
+    jest.resetModules();
+    process.env = env;
+  });
+  describe('getFhirClient', () => {
+    test('throws error when environment variable is missing', async () => {
+      await expect(getFhirClient()).rejects.toThrowError('API_URL environment variable is not defined');
+      process.env.API_URL = 'https://fake-api-url.com/dev';
+      await expect(getFhirClient()).rejects.toThrowError('API_KEY environment variable is not defined');
+      process.env.API_KEY = 'fake-api-key';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'API_AWS_REGION environment variable is not defined'
+      );
+      process.env.API_AWS_REGION = 'us-east-1';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'COGNITO_CLIENT_ID environment variable is not defined'
+      );
+      process.env.COGNITO_CLIENT_ID = 'fake-cognito-client-id';
+      await expect(getFhirClient()).rejects.toThrowError(
+        'COGNITO_USERNAME environment variable is not defined'
+      );
+    });
+
+    test('should call Cognito with correct parameters', async () => {
+      process.env.API_URL = 'https://fake-api-url.com/dev';
+      process.env.API_KEY = 'fake-api-key';
+      process.env.API_AWS_REGION = 'us-east-1';
+      process.env.COGNITO_CLIENT_ID = 'fakeCognitoClientId';
+      process.env.COGNITO_USERNAME = 'fake-cognito-user-name';
```

Comment:
nitpick: you can set `fake-cognito-user-name` to be a variable. Then you can reuse that variable in both the "build" stage and "check" stage.